### PR TITLE
LPD-15145 Clear mapping of DSM fields in cards visualization mode

### PR DESCRIPTION
**Pull request by chestofwonders**
Task: https://liferay.atlassian.net/browse/LPD-15145

**Scope:**

**Allow users to clear and change assigned fields to DSM views in Cards visualization mode**

LPD-10735 FF should be added to access Visualization modes

image

When there is no a field mapped, the "Not Assigned" message should appear:

image

Once a field is mapped, a three-dot button should appears instead of the '+' button:

image

That button opens a dropdown menu that allows user to unmap the field (so the "Not Assigned" label should appear again, and the button should revert to the previous '+') or change the field. In that case, the selection fields modal is open, allowing the user to choose a new field.

image

**Acceptance Criteria:**

- Users can clear the assignment of fields in cards individually.
- When a user clicks on the action, the field is unmapped immediately and persisted.

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx [hunk 1]
```diff
@@ -229,12 +243,13 @@ interface ICardsSectionProps {
...
 	cardsSection,
 	modalProps,
+	onClear,
 	onSelect,
 	saveButtonDisabled,
 }: ICardsSectionProps) {
 	const {field, label} = cardsSection;
 
-	const onClick = () => {
+	const onClickAssign = () => {
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx:252]: I know it is a bit picky :smile:, but wouldn't be clear rename the handler using its purpose, like `handleAssign`, `assignField` as it is already the callback of an `onClick` prop?

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx [hunk 2]
```diff
@@ -257,6 +272,10 @@ function CardsSection({
 		});
 	};
 
+	const onClickClear = () => {
+		onClear(cardsSection);
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx:276]: :sweat_smile: same as above

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx [hunk 3]
```diff
@@ -278,13 +297,55 @@ function CardsSection({
...
 
 					<ClayInput.GroupItem shrink>
-						<ClayButtonWithIcon
-							aria-label={Liferay.Language.get('assign-field')}
-							displayType="secondary"
-							onClick={onClick}
-							symbol="plus"
-							title={Liferay.Language.get('assign-field')}
-						/>
+						{field?.name ? (
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx:300]: Please see [solution from List](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx#L297-L329). The solution in Cards should be nearly the same, only with different labels and CSS classes. We may even consider extracting it to common component(s) in `visualization_modes` scope.

**Review by markocikos — CHANGES_REQUESTED** [review]
@chestofwonders please see comments inline

**Comment by chestofwonders** [comment]
Continues in https://github.com/liferay-frontend/liferay-portal/pull/4051

**Pull request closed without merging**